sql/sql_select.cc

```diff
@@ -675,14 +675,22 @@ setup_without_group(THD *thd, Ref_ptr_array ref_pointer_array,
 
 bool vers_select_conds_t::init_from_sysvar(THD *thd)
 {
-  vers_asof_timestamp_t &in= thd->variables.vers_asof_timestamp;
+  system_variables *vars = &thd->variables;
+  if (vars->vers_asof_timestamp.type == SYSTEM_TIME_UNSPECIFIED)
+    vars = &global_system_variables;
```

Comment:
That's not the way `global_system_variables` are used. Session variable must be used **always**. Global variable is used **only** when new session is created. I.e. session variable is initialized from global variable.